tools/autograd/templates/Functions.cpp

```diff
@@ -274,6 +274,42 @@ Tensor prod_safe_zeros_backward(const Tensor &grad, const Tensor& inp, int64_t d
   return grad * (exclusive_normal * exclusive_reverse);
 }
 
+Tensor nanprod_backward(const Tensor& grad, const Tensor& input, const Tensor& result) {
+  if (input.dim() == 0) {
+    return grad * input.isnan().logical_not();
+  }
+  Tensor zero_idx = (input == 0).nonzero();
+  if (zero_idx.numel() == 0) {
+    return (grad * result * input.isnan().logical_not());
```

Comment:
This formula doesn't look right to me.
Isn't this missing a division by `input`?

Reply:
My bad. Sorry for missing this out, I've committed the change and this should hopefully solve the failures. Thanks @albanD!